test/Microsoft.AspNet.Razor.Test/TagHelpers/TagHelperParseTreeRewriterTest.cs

```diff
@@ -19,6 +19,367 @@ namespace Microsoft.AspNet.Razor.Test.TagHelpers
     public class TagHelperParseTreeRewriterTest : TagHelperRewritingTestBase
     {
         [Fact]
+        public void Rewrite_RecoversWhenRequiredAttributeMismatchAndRestrictedChildren()
```

Comment:
Not seeing any multi-level allowed cases in here. Am I missing something like `<select><option/><optgroup><option/></optgroup></select>` where `<select>` allows either `<option>` or `<optgroup>` and `<optgroup>` allows only `<option>`?